app/controllers/app.py

```diff
@@ -1639,6 +1651,79 @@ def post(self, app_id, revision_id):
             return dict(status='fail', message=str(exc)), 500
 
 
+class AppRedeployView(Resource):
+    @jwt_required
+    def post(self, app_id):
+        """
+        Redeploy application
+        """
+        #TODO: handle private login
+        app_schema = AppSchema()
+        try:
+            current_user_id = get_jwt_identity()
+            current_user_roles = get_jwt_claims()['roles']
+
+            app = App.get_by_id(app_id)
+
+            if not app:
+                return dict(
+                    status="fail",
+                    message=f"App with id {app_id} not found"
+                ), 404
+
+            project = app.project
+
+            if not project:
+                # todo: recreate project
+                return dict(status='fail', message='Internal server error'), 500
+
+            if not is_owner_or_admin(project, current_user_id, current_user_roles):
+                if not is_authorised_project_user(project, current_user_id, 'admin'):
+                    return dict(status='fail', message='Unauthorised'), 403
+
+            cluster = project.cluster
+            namespace = project.alias
+
+            if not cluster or not namespace:
+                return dict(status='fail', message='Internal server error'), 500
+
+            new_app = create_user_app(
+                app,
+                app.alias,
+                app.image,
+                project,
+                command_string=app.command,
+                env_vars=None,
+                private_repo=app.private_image,
+                docker_server=None,
+                docker_username=None,
```

Comment:
@Mubangizi why don't we allow the user to send dockerhub credentials in the request body incase its a private image otherwise this workflow will fail for private images. we can have an optional body that we check if these credentials are sent through.

Reply:
We are not saving these anywhere atm. And I added a TODO comment to handle the private image workflow where we can correctly architect how it will be done. This should be handled next